app/assets/javascripts/discourse/controllers/login.js.es6

```diff
@@ -182,6 +184,29 @@ export default Ember.Controller.extend(ModalFunctionality, {
       const forgotPasswordController = this.get('forgotPassword');
       if (forgotPasswordController) { forgotPasswordController.set("accountEmailOrUsername", this.get("loginName")); }
       this.send("showForgotPassword");
+    },
+
+    emailLogin() {
+      if (Ember.isEmpty(this.get('loginName'))){
+        this.flash(I18n.t('login.blank_username'), 'error');
+        return;
+      }
+
+      ajax('/u/email-login', {
```

Comment:
The button needs to be disabled while the request is being processed. Otherwise, I can spam click the button.

Reply:
Yeah... didn’t seem very bad, given we have rate limiting for this, and worst case you end up with few mails with a link in your inbox, but ok